<launch>
  <!-- Configurable arguments -->
  <arg name="modem_id" default="2"/>
  <arg name="radius" default="10"/>
  <arg name="distance_between_meas" default="1"/>
  <arg name="num_points" default="15"/>
  <arg name="debug" default="False"/>
  <arg name="sim" default="True"/>
  <arg name="interface" default="ethernet"/>
  <arg name="final_depth" default="0"/>
  <arg name="final_altitude" default="10"/>
  <arg name="reference" default="0"/>
  <arg name="heave_mode" default="0"/>
  <arg name="surge_velocity" default="1"/>
  <arg name="tolerance_xy" default="3"/>
  <arg name="timeout" default="1800"/>
  <arg name="no_altitude_goes_up" default="True"/>
  
  <!-- Load services and programs -->
  <node name="go_to_watch_service" pkg="stalker" type="go_to_watch_service.py" args='$(arg final_depth) $(arg final_altitude) $(arg reference) $(arg heave_mode) $(arg surge_velocity) $(arg tolerance_xy) $(arg timeout) $(arg no_altitude_goes_up)' output="screen">
  </node>
  <node name="sim_modem_pos_service" pkg="stalker" type="sim_modem_pos_service.py" args='$(arg debug)' output="screen">
  </node>
  <node name="measure_range_service" pkg="stalker" type="measure_range_service.py" args='$(arg debug) $(arg sim) $(arg interface)' output="screen">
  </node>
  <node name="sim_modem_pos_client" pkg="stalker" type="sim_modem_pos_client.py" args='0 0' output="screen" launch-prefix="bash -c 'sleep 10; $0 $@' ">
  </node>
  <node name="stalker_main" pkg="stalker" type="stalker_main.py" args='$(arg modem_id) $(arg radius) $(arg distance_between_meas) $(arg num_points)' output="screen" launch-prefix="bash -c 'sleep 12; $0 $@' ">
  </node>
</launch>
